consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -691,11 +691,67 @@ impl PruningProofManager {
                     }
                 }
 
+                // Temp assertion for verifying a bug fix: assert that the full 2M chain is actually contained in the composed level proof
+                let set = BlockHashSet::from_iter(headers.iter().map(|h| h.hash));
```

Comment:
Since this uses the same logic as `block_at_depth` wouldn't this function panic before reaching this debug section, making the debug here unreachable for the error it's trying to capture?

Reply:
The flow is (1) build_pruning_point_proof -> (2) prune -> (3) build_pruning_point_proof (sanity). It's at (1) where this is relevant.